src/main/java/cart/controller/product/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package cart.controller.product;
+
+import cart.dto.product.ProductRequest;
+import cart.dto.product.ProductResponse;
+import cart.dto.sale.SaleProductRequest;
+import cart.service.product.ProductService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    public ProductController(final ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ProductResponse>> findAllProducts() {
+        return ResponseEntity.ok(productService.findAllProducts());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ProductResponse> findProductById(@PathVariable("id") final Long productId) {
+        return ResponseEntity.ok(productService.findProductById(productId));
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> createProduct(@RequestBody @Valid final ProductRequest productRequest) {
+        long productId = productService.createProduct(productRequest);
+        return ResponseEntity.created(URI.create("/products/" + productId)).build();
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Void> updateProduct(@PathVariable("id") final Long productId,
+                                              @RequestBody @Valid final ProductRequest productRequest) {
+        productService.updateProduct(productId, productRequest);
+        return ResponseEntity.status(HttpStatus.OK).build();
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteProduct(@PathVariable("id") final Long productId) {
+        productService.deleteProduct(productId);
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    }
+
+    @PostMapping("/sales/{id}")
+    public ResponseEntity<Void> applySale(@PathVariable("id") final Long productId,
+                                          @RequestBody @Valid final SaleProductRequest request) {
+        long id = productService.applySale(productId, request);
+        return ResponseEntity.created(URI.create("/products/" + id)).build();
+    }
+
+    @DeleteMapping("/sales/{id}")
```

Comment:
바로 위 리뷰와 마찬가지로 특정 세일에 대한 id로 보입니다. 변수명 productId를 보아야 비로소 감을 잡을 수 있는데, {productId}로 표시해주는게 백엔드 입장에선 더 나은 표현방식이라고 생각합니다.
 ```suggestion
    @DeleteMapping("/sales/{productId}")
```
그러나 궁극적으로 프론트는 해당 코드를 볼 수 없습니다. 어느정도 공감대가 형성되어 있는 restful API를 통해 쉽게 소통할 수 있도록, `products/{productId}/sales/{saleId}` 혹은 `products/{productId}/sale` 가 좋겠네요.

Reply:
생각없이 기존과 같이 했는데, 새로운 개발자가 협업을 한다고 했을 때, ID가 혼재되어 있으면 헷갈릴 수 있겠네요!

감사합니다 :) 적용되는 부분은 모두 바꾸겠습니다~